pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -97,7 +101,15 @@ func getPortIDByIP(client *gophercloud.ServiceClient, ipAddress string) (string,
 		return true, nil
 	})
 
```

Comment:
(copying comment here from Reviewable weird app output.  I won't use that to add line comments in future :/ )

```go
if err == nil && targetPort == nil {
   err = ErrNotFound
}
```
... otherwise we have a nil port object that no-one is going to check for, leading to runtime crashes :(